### ISSUE TITLE
Installation Error

### ISSUE THREAD

**Issue by airscholar** [sozu-proxy/sozu#908]
## OS Information:
OS: 13.4 Beta (22F5027f)
Chip: Apple M1

## Steps to recreate:
1. curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
2. source "$HOME/.cargo/env"
3. cargo --version
4. cargo install sozu

## Error:
```
error[E0412]: cannot find type `StartupError` in this scope
   --> /Users/hart/.cargo/registry/src/github.com-1ecc6299db9ec823/sozu-0.14.2/src/main.rs:297:52
    |
297 | fn update_process_limits(_: &Config) -> Result<(), StartupError> {
    |                                                    ^^^^^^^^^^^^ not found in this scope
    |
help: you might be missing a type parameter
    |
297 | fn update_process_limits<StartupError>(_: &Config) -> Result<(), StartupError> {
    |                         ++++++++++++++

error[E0277]: the `?` operator can only be applied to values that implement `Try`
   --> /Users/hart/.cargo/registry/src/github.com-1ecc6299db9ec823/sozu-0.14.2/src/command/mod.rs:302:40
    |
302 |         let executable_path = unsafe { get_executable_path()? };
    |                                        ^^^^^^^^^^^^^^^^^^^^^^ the `?` operator cannot be applied to type `std::string::String`
    |
    = help: the trait `Try` is not implemented for `std::string::String`

error[E0277]: the `?` operator can only be applied to values that implement `Try`
   --> /Users/hart/.cargo/registry/src/github.com-1ecc6299db9ec823/sozu-0.14.2/src/command/mod.rs:481:40
    |
481 |         let executable_path = unsafe { get_executable_path()? };
    |                                        ^^^^^^^^^^^^^^^^^^^^^^ the `?` operator cannot be applied to type `std::string::String`
    |
    = help: the trait `Try` is not implemented for `std::string::String`

error[E0599]: no method named `with_context` found for struct `String` in the current scope
   --> /Users/hart/.cargo/registry/src/github.com-1ecc6299db9ec823/sozu-0.14.2/src/main.rs:152:47
    |
152 | ...table_path().with_context(|| "Could not get executable path")? };
    |                 ^^^^^^^^^^^^ method not found in `String`

Some errors have detailed explanations: E0277, E0412, E0599.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `sozu` due to 4 previous errors
error: failed to compile `sozu v0.14.2`, intermediate artifacts can be found at `/var/folders/yn/vhctpg1507j7b32gmsyq2tkh0000gp/T/cargo-installfvccc8`
```

**Comment by Keksoj** [comment]
Hello,
I'm sorry this happened to you, it is a known bug and was identified in this PR:

- #869 

Unfortunately the fix to build for MacOS is not present in version `0.14.2`. What you can do, however, is build the latest version of the code by pulling from github directly.

```
git clone git@github.com:sozu-proxy/sozu.git
cd sozu
cargo install .
```

@FlorentinDUBOIS how about we publish a new release that contains this fix?

**Comment by FlorentinDUBOIS** [comment]
Hello, the fix has been released in the v0.14.3 and v0.15.0. I close the issue.
